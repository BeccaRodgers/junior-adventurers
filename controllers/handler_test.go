package controllers

import (
	approvals "github.com/approvals/go-approval-tests"
	"github.com/stretchr/testify/assert"
	"net/http"
	"net/http/httptest"
	"testing"
)

func TestHandler_Homepage(t *testing.T) {
	approveGetPage(t, "/")
}

func TestHandler_GuildPage(t *testing.T) {
	approveGetPage(t, "/guilds/1")
}

func approveGetPage(t *testing.T, path string) {
	t.Helper()

	handler := Handler()
	request := httptest.NewRequest(http.MethodGet, path, nil)
	recorder := httptest.NewRecorder()
	handler.ServeHTTP(recorder, request)
	assert.Equal(t, http.StatusOK, recorder.Code)
	approvals.Verify(t, recorder.Body)
}
