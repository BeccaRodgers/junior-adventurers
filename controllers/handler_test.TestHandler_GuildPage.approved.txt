<!doctype html><html lang="en"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta http-equiv="X-UA-Compatible" content="ie=edge"><title>Junior Adventurers</title><link rel="stylesheet" href="/static/style.css"><script src="/static/htmx.min.js"></script></head><body hx-target="this"><section class="section"><div class="container has-text-centered"><h1 class="title">Founders Guild</h1></div><div class="container"><h2 class="subtitle has-text-centered">Leaders</h2><div class="card"><div class="card-content"><p class="title">Beyoncé</p><p class="subtitle">GuildMaster</p></div></div><table class="table table is-bordered is-striped is-hoverable is-fullwidth"><thead><tr><th>Name</th><th>Species</th></tr></thead> <tbody><tr><td>Alex</td><td>Human</td></tr><tr><td>Beyoncé</td><td>Dwarf</td></tr></tbody></table><h2 class="subtitle has-text-centered">Members</h2><table class="table table is-bordered is-striped is-hoverable is-fullwidth"><thead><tr><th>Name</th><th>Species</th><th>Age</th></tr></thead> <tbody><tr><td>Angela</td><td>Human</td><td>10</td></tr><tr><td>Bob</td><td>Human</td><td>12</td></tr></tbody></table><button class="button is-link" hx-get="/" hx-push-url="true">Home</button></div></section></body></html>